libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -613,4 +621,7 @@ namespace breez_sdk {
  sequence<u8> mnemonic_to_seed(string phrase);
 
  Config default_config(EnvironmentType env_type, string api_key, NodeConfig node_config);
+
+ [Throws=SdkError]
+ StaticBackupResponse static_backup(StaticBackupRequest request);
```

Comment:
Is there a reason static_backup is exposed as a "static" method of the SDK instead of a service instance method?
These are my two concerns:

1. I think our intention with handling multiple instance states is for the connect method to create/determine a subdirectory off of the base working directory derived from the node seed. With this the client does not need to manage which working_dir is used for which seed. A side effect of this is the client using the SDK will not know the working_dir of the node (only the base working directory). If this is a instance method, then the working_dir is not needed.
2. Although this does give access to the backup while the node is not connected, it could lead to the client using the SDK to download an out-of-date backup. If this is a instance method and the sync had occurred, then at least we know we are returning the latest static backup.

Reply:
The requirements for the whole feature is to be able to access the static backup data when the sdk is not able to connect to the node (sort of doom day when greenlight is down for example).
To meet this requirement we can't count on connect to succeed and this is the reason I chose a static API.
As for multiple nodes I think this can evolve to that. The working_dir (base directory) would still be needed but when we introduce multiple nodes under this need we might need to add another field to the StaticBackupRequest such as the seed or the node id.
As for the latest file, we do fetch it on every new channel (see the sync logic) so even if called statically it should be up to date.
LMK if you see any better option here.

Reply:
> The requirements for the whole feature is to be able to access the static backup data when the sdk is not able to connect to the node (sort of doom day when greenlight is down for example).
> To meet this requirement we can't count on connect to succeed and this is the reason I chose a static API.
> As for multiple nodes I think this can evolve to that. The working_dir (base directory) would still be needed but when we introduce multiple nodes under this need we might need to add another field to the StaticBackupRequest such as the seed or the node id.

👍 
> As for the latest file, we do fetch it on every new channel (see the sync logic) so even if called statically it should be up to date.

I was just thinking of a scenario where multiple clients/app connect to the same node. If Client A opens a new channel, the static backup of Client A is synced. Any other clients have stale static backups until they connect and sync. These stale backups could potentially be used to recover from. I guess there is not much you can do to prevent this, if its the dooms day scenario.

Reply:
I agree there is no 100% solution here especially if we have multiple clients. In this case there are highly chances that at list one of them will have the most up to static backup data.
Even with one instance there could be a race where the sync failed or some network error happened.
I think we can say this thing for sure. As long as the sync API call succeed then the static data should be up to date.